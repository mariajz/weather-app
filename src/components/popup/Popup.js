import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import Modal from 'react-native-modal';
import { CustomIcon } from '../../commons/visual-elements';
import { Content, ContentWrapper, StyledTitle } from './Popup.style';

const Popup = ({ popupProps }) => {
    const { onClose, title, description } = popupProps;

    return (
        <Modal
            isVisible
            backdropColor="rgba(0,0,0,0.5)"
            animationIn="fadeIn"
            animationOut="fadeOut">
            <ContentWrapper>
                <Content>
                    <StyledTitle>{title}</StyledTitle>
                    <Text>{description}</Text>
                </Content>
                <TouchableOpacity onPress={onClose}>
                    <CustomIcon size={15} iconName="close" tintColor="red" />
                </TouchableOpacity>
            </ContentWrapper>
        </Modal>
    );
};

export default Popup;
