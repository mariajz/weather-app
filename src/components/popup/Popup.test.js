import { render } from '@testing-library/react-native';
import React from 'react';
import { View as MockView } from 'react-native';
import Popup from './Popup';

const mockOnClose = jest.fn();
const popupProps = {
    description: 'desc',
    onClose: mockOnClose,
    title: 'Error',
};

jest.mock('../../commons/visual-elements', () => ({
    CustomIcon: () => <MockView />,
}));

jest.mock('react-native-modal', () => props => <MockView {...props} />);

describe('Popup', () => {
    it('should render Popup', () => {
        const container = render(<Popup popupProps={popupProps} />);

        expect(container).toMatchSnapshot();
    });
});
