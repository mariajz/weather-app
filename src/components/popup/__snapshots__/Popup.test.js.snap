// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup should render Popup 1`] = `
<View
  animationIn="fadeIn"
  animationOut="fadeOut"
  backdropColor="rgba(0,0,0,0.5)"
  isVisible={true}
>
  <View
    style={
      {
        "backgroundColor": "white",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "flexBasis": 0,
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-between",
        "maxHeight": 100,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        Error
      </Text>
      <Text>
        desc
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View />
    </View>
  </View>
</View>
`;
